src/renderer.hpp

```diff
@@ -45,15 +45,15 @@ namespace wr
 
 		virtual void WaitForAllPreviousWork() = 0;
 
-		virtual CommandList* GetDirectCommandList(unsigned int num_allocators) = 0;
-		virtual CommandList* GetBundleCommandList(unsigned int num_allocators) = 0;
-		virtual CommandList* GetComputeCommandList(unsigned int num_allocators) = 0;
-		virtual CommandList* GetCopyCommandList(unsigned int num_allocators) = 0;
-		virtual RenderTarget* GetRenderTarget(RenderTargetProperties properties) = 0;
-		virtual void ResizeRenderTarget(RenderTarget** render_target, std::uint32_t width, std::uint32_t height) = 0;
+		virtual CommandList GetDirectCommandList(unsigned int num_allocators) = 0;
```

Comment:
This is a perfect example of why typedefing CommandList and RenderTarget to void* can cause confusion in reading the code. It was our mistake. It's better to typedef CommandList to void so that we can use CommandList* as a return type in example and make sure the users see that it's a pointer